addons/l10n_id_efaktur_coretax/models/account_move_line.py

```diff
@@ -52,4 +53,10 @@ def _l10n_id_coretax_build_invoice_line_vals(self, vals):
         line_val['VAT'] = idr.round(line_val['OtherTaxBase'] * line_val['VATRate'] / 100)
         line_val['STLG'] = idr.round(line_val['STLGRate'] * line_val['OtherTaxBase'] / 100)
 
+        # for numerical attributes in line_val, use float_repr to ensure proper formattinf
+        numerical_fields = ['Price', 'TotalDiscount', 'TaxBase', 'OtherTaxBase', 'VAT', 'STLG']
+        for field in numerical_fields:
+            line_val[field] = float_repr(line_val[field], precision_digits=self.currency_id.decimal_places)
```

Comment:
Be careful that float_repr **does not round**, it only format the value.
It's not an issue here for most fields; as you do idr.round before.

But you don't do it for the Price in the vals, you need to add it there too first to be sure it is rounded properly; then formatted to remove any rounding issue.